tests/db_level_on_delete/tests.py

```diff
@@ -0,0 +1,171 @@
+from django.db import IntegrityError, transaction
```

Comment:
I'd move these tests to the `tests/delete/` and reuse existing models (when appropriate).

Reply:
Done.